src/ui/common/src/components/pages/workflows/index.tsx

```diff
@@ -66,34 +61,122 @@ const WorkflowsPage: React.FC<Props> = ({ user, Layout = DefaultLayout }) => {
     </Typography>
   );
 
-  const workflowList = filteredList(
-    filterText,
-    allWorkflows.workflows,
-    getOptionLabel,
-    displayFilteredWorkflows,
-    noItemsMessage
-  );
+  const workflows = allWorkflows.workflows;
+
+  /**
+   * Iterate through workflows array and map each element to a WorkflowTableRow object.
+   */
+  const workflowElements: PaginatedSearchTableRow[] = workflows.map((value) => {
+    const engineName =
+      value.engine[0].toUpperCase() + value.engine.substring(1);
+    const engineIconUrl =
+      ServiceLogos[value.engine[0].toUpperCase() + value.engine.substring(1)];
+
+    const metrics = value.metrics.map((metric) => {
+      return {
+        metricId: metric.id,
+        name: metric.name,
+        value: metric.result.content_serialized,
+        status: metric.result.exec_state.status,
+      };
+    });
+
+    const checks = value.checks.map((check) => {
+      const value =
+        check.result.exec_state.status === 'succeeded' ? 'True' : 'False';
+
+      return {
+        checkId: check.id,
+        name: check.name,
+        level: check.spec.check.level,
+        timestamp: check.result.exec_state.timestamps.finished_at,
+        value,
+        status: check.result.exec_state.status,
+      };
+    });
+
+    const workflowTableRow: PaginatedSearchTableRow = {
+      name: {
+        name: value.name,
+        url: `/workflow/${value.id}`,
+        status: value.status,
+      },
+      last_run: new Date(value.last_run_at * 1000).toLocaleString(),
+      engine: {
+        engineName,
+        engineIconUrl: engineIconUrl,
+      },
+      // TODO: add metrics and checks to response item when getting workflows list.
+      metrics,
+      checks,
+    };
+
+    return workflowTableRow;
+  });
+
+  const workflowTableData: PaginatedSearchTableData = {
+    schema: {
+      fields: [
+        { name: 'name', type: 'varchar' },
+        { name: 'last_run', displayName: 'Last Run', type: 'varchar' },
+        { name: 'engine', type: 'varchar' },
+        { name: 'metrics', type: 'varchar' },
+        { name: 'checks', type: 'varchar' },
+      ],
+      pandas_version: '1.5.1',
```

Comment:
Is this just an example? Why is this fixed at `1.5.1`?

Reply:
this field technically isn't used in the table @cw75 I'll make a task to remove it.